rfcs/rfc-metaprogramming.md

```diff
@@ -0,0 +1,379 @@
+# Metaprogramming API
+
+## Preamble
+
+    Author:  Paul Evans <PEVANS>
+    Sponsor:
+    ID:      TODO
+    Status:  Exploratory
+
+## Abstract
+
+A coherent API for metaprogramming tasks; that is, code which is aware of its own code structure and can inspect or manipulate it.
+
+## Motivation
+
+As a dynamic language, programs written in Perl have the ability to be aware of the structure of their own code. Functions, packages, variables, and other constructions can be obtained or manipulated by name, While the techniques can certainly be over-used, it is a useful and necessary ability of what it means to be a "dynamic language", and where appropriate these kinds of techniques should be supported and encouraged.
```

Comment:
nit: `...name, While the techniques...` should the comma here be a period or the while lowercased...